spec.html

```diff
@@ -49014,7 +49014,7 @@ <h2>Syntax</h2>
           [+UnicodeMode] Assertion[+UnicodeMode, ?UnicodeSetsMode, ?NamedCaptureGroups]
```

Comment:
It's also possible to reduce branching in this production, but I'm not sure it increases clarity as it mixes the `+UnicodeMode` and `~UnicodeMode` branches:

```
        Term[UnicodeMode, UnicodeSetsMode, NamedCaptureGroups] ::
          [~UnicodeMode] LookaheadAssertion[~UnicodeMode, ~UnicodeSetsMode, ?NamedCaptureGroups] Quantifier
          Assertion[?UnicodeMode, ?UnicodeSetsMode, ?NamedCaptureGroups]
          [~UnicodeMode] ExtendedAtom[?NamedCaptureGroups] Quantifier?
          [+UnicodeMode] Atom[+UnicodeMode, ?UnicodeSetsMode, ?NamedCaptureGroups] Quantifier?
```

It could probably be further simplified by merging `ExtendedAtom` with `Atom` and moving the branching on UnicodeMode inside it, but the refactor becomes larger:
```
        Term[UnicodeMode, UnicodeSetsMode, NamedCaptureGroups] ::
          [~UnicodeMode] LookaheadAssertion[~UnicodeMode, ~UnicodeSetsMode, ?NamedCaptureGroups] Quantifier
          Assertion[?UnicodeMode, ?UnicodeSetsMode, ?NamedCaptureGroups]
          Atom[?UnicodeMode, ?UnicodeSetsMode, ?NamedCaptureGroups] Quantifier?

        Atom[UnicodeMode, UnicodeSetsMode, NamedCaptureGroups] ::
          `.`
          `\` AtomEscape[?UnicodeMode, ?NamedCaptureGroups]
          [~UnicodeMode] `\` [lookahead == `c`]
          CharacterClass[?UnicodeMode, ?UnicodeSetsMode]
          `(` GroupSpecifier[?UnicodeMode]? Disjunction[?UnicodeMode, ?UnicodeSetsMode, ?NamedCaptureGroups] `)`
          `(?:` Disjunction[?UnicodeMode, ?UnicodeSetsMode, ?NamedCaptureGroups] `)`
          [~UnicodeMode] InvalidBracedQuantifier
          [~UnicodeMode] ExtendedPatternCharacter
          [+UnicodeMode] PatternCharacter
```